sdk/rust/oak/src/lib.rs

```diff
@@ -37,7 +37,16 @@ use serial_test_derive::serial;
 /// Handle used to identify read or write channel halves.
 ///
 /// These handles are used for all host function calls.
-pub type Handle = u64;
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct Handle {
+    id: u64,
+}
+
+impl Handle {
+    pub fn new (id: u64) -> Handle {
```

Comment:
Maybe call this `from_raw` or similar? And also document that this should (usually) not be called from user code, but from serialisation and other internal libraries.